pacai/util/probability.py

```diff
@@ -4,27 +4,23 @@
 
 import random
 
-from pacai.util import counter
-
-def normalize(vectorOrCounter):
+def normalize(vectorOrDict):
     """
-    Normalize a vector or counter by dividing each value by the sum of all values.
+    Normalize a vector or dictionary by dividing each value by the sum of all values.
     """
 
-    normalizedCounter = counter.Counter()
-    if type(vectorOrCounter) == type(normalizedCounter):
-        counterContainer = vectorOrCounter
-        total = float(counterContainer.totalCount())
+    if type(vectorOrDict) == dict:
```

Comment:
Instead of doing a strict type check here, do an instance check:
```
isinstance(vectorOrCounter, dict)
```

We want to allow people to also pass a subclass of `dict`.

Reply:
This is not resolved.
`type(vectorOrDict) == dict` is still a "strict" type check, you are making sure it is only a specific type.
That is if it is something that extends `dict`?

Use:
```
isinstance(vectorOrDict, dict)
```
Then, it only has to be an instance (or subclass) of `dict`.